content/en/continuous_integration/intelligent_test_runner/how_it_works.md

```diff
@@ -0,0 +1,61 @@
+
+Other test impact analysis solutions are based off code coverage but are hard to set up. Additionally, they only look at the diff in your latest commit to evaluate which tests to run. This means that you need to guarantee that all your commits run through CI. Even if you do, this doesn't play well with GitHub's Pull Requests, which only take into account the CI status of the latest commit to allow merging the PR.
+
+Intelligent Test Runner leverages per-test code coverage information along with data from [Test Visibility](../../tests/) to search previous test in all relevant past commits. Configuration of Intelligent Test Runner is a one-click operation on most languages, and the results are accurate and more precise than other methods.
+
+
+## How test selection works
+
+When you enable Intelligent Test Runner, per-test (or per-suite, depending on the framework) code coverage is transparently collected and sent to Datadog.
+
+When starting a new test session in your CI, the Datadog library requests a list of skippable tests from the backend. This request includes information about the current commit, the service that is being tested, and OS and runtime configuration. The Datadog backend uses that information to search for previous test runs that can be used as proof that, if a specific test is run, we wouldn't be gaining any value. This happens whenever a previous passed test run is found that was run on a commit that has the same version of the covered and [tracked](../#tracked-files) files.
+
+{{< img src="continuous_integration/itr_test_selection.png" alt="Intelligent Test Runner Test Selection Diagram" style="width:60%;">}}
+
+The Datadog library then removes tests marked as unskippable in source from the skippable tests list. It then proceeds to run the tests, but directs the test framework to skip those that remain in the skippable test list.
+
+{{< img src="continuous_integration/itr_skipped_test_run.png" alt="Intelligent Test Runner Skipped Test" style="width:80%;">}}
+
+Let's take a look at a specific example:
+
+{{< img src="continuous_integration/itr_example.png" alt="Intelligent Test Runner Example Diagram" style="width:80%;">}}
+
+In the previous diagram shows a developer branch that branches out from `main` and has several commits. On each commit, the CI has been running two tests (A and B) with different results.
+
+- **Commit 1** ran both tests. This commit contained changes that affected the tracked files, and the covered files of both A and B.
+- **Commit 2** ran both tests again:
+  - Test A has to be run because, although this commit did not affect test A (no changes in tracked files or covered files), there are no previous test runs that passed for Test A. Since Intelligent Test Runner cannot guarantee that the test will pass if were run, it doesn't skip it. This time, the test passes, which indicates this is a flaky test.
```

Comment:
```suggestion
  - Test A has to be run because, although this commit did not affect test A (no changes in tracked files or covered files), there are no previous test runs that passed for Test A. Since Intelligent Test Runner cannot guarantee that the test will pass if it were run, it doesn't skip it. This time, the test passes, which indicates this is a flaky test.
```

Reply:
Addressed this sentence along with the use of "will"